### PR TITLE
Fix issues to remove the need for wait

### PR DESCRIPTION
**Pull request by antoyo**

### CODE REVIEW

**Comment by antoyo** [comment]
cc @GuillaumeGomez: I think it's ready to be review/merged.

File: src/functions.rs [hunk 1]
```diff
@@ -57,13 +64,25 @@ use gtk::{
...
 /// # }
 /// ```
-pub fn click<W: Clone + IsA<Object> + IsA<Widget> + WidgetExt>(widget: &W) {
+pub fn click<W: Clone + IsA<Object> + IsA<Widget> + WidgetExt + IsA<W>>(widget: &W) {
     wait_for_draw(widget, || {
+        let observer =
+            if let Ok(tool_button) = widget.clone().dynamic_cast::<ToolButton>() {
+                observer_new!(tool_button, connect_clicked, |_|)
+            }
+            else if let Ok(tool_button) = widget.clone().dynamic_cast::<Button>() {
```
Comment (GuillaumeGomez) [inline src/functions.rs:73]: Please run `cargo fmt`. :)

**Comment by antoyo** [comment]
@GuillaumeGomez: Fixed.

**Comment by GuillaumeGomez** [comment]
Thanks! Please ping me once CIs pass. :)

**Pull request merged**